pkg/vz/network_darwin.go

```diff
@@ -98,14 +98,7 @@ func (c *qemuPacketConn) Read(b []byte) (n int, err error) {
 		// Likely connection closed by peer.
 		return 0, err
 	}
-
-	reader := io.LimitReader(c.Conn, int64(size))
-	_, err = reader.Read(b)
-	if err != nil {
-		// Likely connection closed by peer.
-		return 0, err
-	}
-	return int(size), nil
+	return io.ReadFull(c.Conn, b[:size])
```

Comment:
We indeed need to do a full read here, so this change looks correct. The current code may do a short read and return size without checking the amount of data read. This will return a packet with partial data and random junk at end of the buffer left from a previous read.